holidays/countries/nigeria.py

```diff
@@ -25,82 +26,98 @@ class Nigeria(HolidayBase):
     """
 
     country = "NG"
+    special_holidays = {
+        2019: (
+            (FEB, 22, "Public Holiday for Elections"),
+            (MAY, 29, "Presidential Inauguration Day"),
+        ),
+    }
 
     def _populate(self, year):
-        def _add_holiday(dt: date, hol: str) -> None:
-            """Only add if in current year; prevents adding holidays across
-            years (handles multi-day Islamic holidays that straddle
-            Gregorian years).
-            """
-            if dt.year == year:
-                self[dt] = hol
+        def collect_holiday(hol_date: date, hol_name: str) -> None:
+            if hol_date in holiday_date_names_mapping:
+                holiday_date_names_mapping[hol_date].append(hol_name)
+            else:
+                holiday_date_names_mapping[hol_date] = [hol_name]
 
         if year <= 1978:
             return None
 
         super()._populate(year)
 
+        holiday_date_names_mapping: Dict[date, List[str]] = {}
+
         # New Year's Day
-        self[date(year, JAN, 1)] = "New Year's day"
+        collect_holiday(date(year, JAN, 1), "New Year's Day")
 
-        # Calculate Easter for given year
-        # followed by easter related holidays
         easter_date = easter(year)
-        self[easter_date + td(days=-2)] = "Good Friday"
-        self[easter_date + td(days=+1)] = "Easter Monday"
+        collect_holiday(easter_date + td(days=-2), "Good Friday")
+        collect_holiday(easter_date + td(days=+1), "Easter Monday")
 
         # Worker's day
         if year >= 1981:
-            self[date(year, MAY, 1)] = "Workers' day"
+            collect_holiday(date(year, MAY, 1), "Workers' Day")
+
+        # Democracy day moved around after its inception in 2000
+        # Initally it fell on May 29th
+        # In 2018 it was announced that the holiday
+        # will move to June 12th from 2019
+        if year >= 2000:
+            collect_holiday(
+                date(year, JUN, 12) if year >= 2019 else date(year, MAY, 29),
+                "Democracy Day",
+            )
+
+        # Independence Day
+        collect_holiday(date(year, OCT, 1), "Independence Day")
+
+        # Christmas day
+        collect_holiday(date(year, DEC, 25), "Christmas Day")
+
+        # Boxing day
+        collect_holiday(date(year, DEC, 26), "Boxing Day")
 
         # Eid al-Fitr - Feast Festive
         # This is an estimate
         # date of observance is announced yearly
         for yr in (year - 1, year):
             for hol_date in _islamic_to_gre(yr, 10, 1):
-                _add_holiday(hol_date, "Eid al-Fitr")
-                _add_holiday(hol_date + td(days=+1), "Eid al-Fitr Holiday")
+                collect_holiday(hol_date, "Id el Fitr")
```

Comment:
Why these new names for Eid al-Fitr and Eid al-Adha? What language is that?

Reply:
On https://interior.gov.ng also use [Eidul-Fitr](https://interior.gov.ng/fg-declares-wednesday-12th-thursday-13th-may-2021-as-public-holidays-to-mark-eidul-fitr-celebration/), [Eidul-Mawlid](https://interior.gov.ng/federal-government-has-declared-thursday-october-29-2020-as-a-public-holiday-to-mark-the-eidul-mawlid/), [Eid-el-Kabir/Eid-el Kabir](https://interior.gov.ng/fg-declares-july-11th-and-12th-2022-as-public-holiday-to-mark-eid-el-kabir-celebration/)... So I don't even know which option is more correct. :(

Just for consistency I choose Eid-el-Fitr, Eid-el-Kabir, Eid-el-Mawlid, and I'll waiting for your opinion.